src/Bicep.Core/Configuration/ConfigHelper.cs

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Configuration;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Configuration
+{
+    internal class ConfigHelper
+    {
+        private const string SettingsFileName = "bicepsettings.json";
+
+        /// <summary>
+        /// Property exposes the configuration root
+        /// that is currently loaded
+        /// </summary>
+        public IConfigurationRoot Config { get; private set; }
+
+        internal ConfigHelper()
+        {
+            this.Config = BuildConfig(Directory.GetCurrentDirectory());
+        }
+
+        private IConfigurationRoot BuildConfig(string localFolder)
+        {
+            var configBuilder = new ConfigurationBuilder();
+
+            // load the default settings from file embedded as resource
+            var assembly = Assembly.GetExecutingAssembly();
+            var names = assembly.GetManifestResourceNames();
+            var defaultConfigResourceName = names.FirstOrDefault(n => n.EndsWith(SettingsFileName));
+
+            // keep this stream open until after Build() call
+            using (var defaultConfigStream = assembly.GetManifestResourceStream(defaultConfigResourceName))
+            {
+                Debug.Assert(defaultConfigStream != null, "Default configuration file should exist as embedded resource.");
+                configBuilder.AddJsonStream(defaultConfigStream);
+
+                // last added json settings take precedent - add local settings last
+                if (DiscoverLocalConfigurationFile(localFolder) is string localConfig)
+                {
+                    configBuilder.AddJsonFile(localConfig, true, true);
+                    this.CustomSettingsFileName = localConfig;
+                }
+                else
+                {
+                    this.CustomSettingsFileName = default;
+                }
+
+                return configBuilder.Build();
+            }
+        }
+
+        private string? DiscoverLocalConfigurationFile(string? nextDir)
+        {
+            while (nextDir != default)
+            {
+                var fileName = Path.Combine(nextDir, SettingsFileName);
+                if (File.Exists(fileName))
+                {
```

Comment:
File.Exists() can throw for various reasons including no having access to root folders.  You should ignore System.IO-derived exceptions at least.  Probably should also ignore others, but shoot them out to telemetry.

Reply:
Must have for MVP

Reply:
Right now this is what we see if the settings file parsing fails:

![image](https://user-images.githubusercontent.com/6913354/119034092-da940780-b962-11eb-928d-7ec12ecb1edd.png)

Reply:
https://github.com/jfleisher/bicep/pull/14